[CI/CD] Build failure due to "No POM"

I'm trying to get a very simple HelloWorld WildFly example to build in fabric8 using the CI/CD pipeline. My environment is a local Vagrant installation on OS X.

When I'm doing this manually using the following command:
`mvn install docker:build fabric8:json fabric8:apply`
everything works perfectly.

However when trying to import my git repo as an app in fabric8 and triggering a build I get the following error (snippet):

> 
> mvn org.codehaus.mojo:versions-maven-plugin:2.2:set -DnewVersion=1.0.5
> [INFO] Building Maven Stub Project (No POM) 1
> [ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:set (default-cli): Goal requires a project to execute but there is no POM in this directory (/home/jenkins/workspace/workspace/fabric8-wildfly-helloworld). Please verify you invoked Maven from the correct directory.

(full log: [build_log_no_pom.txt](https://github.com/fabric8io/fabric8/files/210446/build_log_no_pom.txt))

Obviously, my project has a pom.xml but it doesn't seem to be able to find it.

What am I doing wrong?

Ah I saw this yesterday on one of our environments.  At a guess until I can have a proper look, I wonder if you selected the option to copy the pipeline to the project workspace? There's a checkbox under the pipeline selection options. 

If you're able to give that a try it might be the problem, if it is we'll put a proper fix together. 

James. 

> On 8 Apr 2016, at 17:02, Sander Knopper <notifications@github.com> wrote:
> 
> I'm trying to get a very simple HelloWorld WildFly example to build in fabric8 using the CI/CD pipeline. My environment is a local Vagrant installation on OS X.
> 
> When I'm doing this manually using the following command:
> mvn install docker:build fabric8:json fabric8:apply
> everything works perfectly.
> 
> However when trying to import my git repo as an app in fabric8 and triggering a build I get the following error (snippet):
> 
> mvn org.codehaus.mojo:versions-maven-plugin:2.2:set -DnewVersion=1.0.5
> [INFO] Building Maven Stub Project (No POM) 1
> [ERROR] Failed to execute goal org.codehaus.mojo:versions-maven-plugin:2.2:set (default-cli): Goal requires a project to execute but there is no POM in this directory (/home/jenkins/workspace/workspace/fabric8-wildfly-helloworld). Please verify you invoked Maven from the correct directory.
> 
> (full log: build_log_no_pom.txt)
> 
> Obviously, my project has a pom.xml but it doesn't seem to be able to find it.
> 
> What am I doing wrong?
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>

--- a/workflows/django/CanaryReleaseStage/Jenkinsfile
+++ b/workflows/django/CanaryReleaseStage/Jenkinsfile
@@ -3,7 +3,7 @@ def utils = new io.fabric8.Utils()
 
 node {
   def envStage = utils.environmentNamespace('staging')
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/django/CanaryReleaseStageAndApprovePromote/Jenkinsfile
+++ b/workflows/django/CanaryReleaseStageAndApprovePromote/Jenkinsfile
@@ -5,7 +5,7 @@ node {
   def envStage = utils.environmentNamespace('staging')
   def envProd = utils.environmentNamespace('production')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/golang/CanaryReleaseStage/Jenkinsfile
+++ b/workflows/golang/CanaryReleaseStage/Jenkinsfile
@@ -4,7 +4,7 @@ def utils = new io.fabric8.Utils()
 node {
   def envStage = utils.environmentNamespace('staging')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/golang/CanaryReleaseStageAndApprovePromote/Jenkinsfile
+++ b/workflows/golang/CanaryReleaseStageAndApprovePromote/Jenkinsfile
@@ -5,7 +5,7 @@ node {
   def envStage = utils.environmentNamespace('staging')
   def envProd = utils.environmentNamespace('production')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/maven/CanaryReleaseAndStage/Jenkinsfile
+++ b/workflows/maven/CanaryReleaseAndStage/Jenkinsfile
@@ -26,7 +26,7 @@ def utils = new io.fabric8.Utils()
 node {
   def envStage = utils.environmentNamespace('staging')
 
-  checkout scm
+  git GIT_URL
 
   kubernetes.pod('buildpod').withImage('fabric8/maven-builder')
       .withPrivileged(true)

--- a/workflows/maven/CanaryReleaseStageAndApprovePromote/Jenkinsfile
+++ b/workflows/maven/CanaryReleaseStageAndApprovePromote/Jenkinsfile
@@ -29,7 +29,7 @@ node {
   def envStage = utils.environmentNamespace('staging')
   def envProd = utils.environmentNamespace('production')
 
-  checkout scm
+  git GIT_URL
 
   kubernetes.pod('buildpod').withImage('fabric8/maven-builder')
       .withPrivileged(true)

--- a/workflows/node/CanaryReleaseStage/Jenkinsfile
+++ b/workflows/node/CanaryReleaseStage/Jenkinsfile
@@ -4,7 +4,7 @@ def utils = new io.fabric8.Utils()
 node {
   def envStage = utils.environmentNamespace('staging')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/node/CanaryReleaseStageAndApprovePromote/Jenkinsfile
+++ b/workflows/node/CanaryReleaseStageAndApprovePromote/Jenkinsfile
@@ -5,7 +5,7 @@ node {
   def envStage = utils.environmentNamespace('staging')
   def envProd = utils.environmentNamespace('production')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/rails/CanaryReleaseStage/Jenkinsfile
+++ b/workflows/rails/CanaryReleaseStage/Jenkinsfile
@@ -4,7 +4,7 @@ def utils = new io.fabric8.Utils()
 node {
   def envStage = utils.environmentNamespace('staging')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/rails/CanaryReleaseStageAndApprovePromote/Jenkinsfile
+++ b/workflows/rails/CanaryReleaseStageAndApprovePromote/Jenkinsfile
@@ -5,7 +5,7 @@ node {
   def envStage = utils.environmentNamespace('staging')
   def envProd = utils.environmentNamespace('production')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/swift/CanaryReleaseStage/Jenkinsfile
+++ b/workflows/swift/CanaryReleaseStage/Jenkinsfile
@@ -4,7 +4,7 @@ def utils = new io.fabric8.Utils()
 node {
   def envStage = utils.environmentNamespace('staging')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

--- a/workflows/swift/CanaryReleaseStageAndApprovePromote/Jenkinsfile
+++ b/workflows/swift/CanaryReleaseStageAndApprovePromote/Jenkinsfile
@@ -5,7 +5,7 @@ node {
   def envStage = utils.environmentNamespace('staging')
   def envProd = utils.environmentNamespace('production')
 
-  checkout scm
+  git GIT_URL
 
   stage 'Canary release'
   if (!fileExists ('Dockerfile')) {

